### DISCUSSION TITLE
Is it possible to force-ignore numbers to be treated as a parameter in a scenario?

### DISCUSSION THREAD

**Discussion by mfelipe** [cucumber/godog#559] [Q&A]
Hello all!

I have the following example scenario (sorry for the excess of creativity):

```gherkin
    Scenario: All your base10s are belong to ustring
        Given a scenario I want to test
        When I run it with a number1 which should not be a "parameter"
        Then godog is a badog
``` 
When I run godog, it suggests the following
```go
func iRunItWithANumberWhichShouldNotBeA(arg1 int, arg2 string) error {
	return godog.ErrPending
}

func InitializeScenario(ctx *godog.ScenarioContext) {
	ctx.Step(`^I run it with a number(\d+) which should not be a "([^"]*)"$`, iRunItWithANumberWhichShouldNotBeA)
}
``` 
I know that I can simply change "number(\d+)" to "number1" at the step and remove the int parameter, but that's not my point. Is there a native way of making godog ignore expressions containing numbers to avoid this numbers to be treated as numeric parameters?

Cheers!

**Comment by vearutop** [comment]
This question is more about regular expressions, `godog` takes arguments that are captured by regexp, regexp captures parts that are enclosed in `()`.

So I think in your case you just need to remove `()` around number.

```go
func iRunItWithANumberWhichShouldNotBeA(arg2 string) error {
	println(arg2)
	return nil
}

func InitializeScenario(ctx *godog.ScenarioContext) {
	ctx.Step(`^I run it with a number\d+ which should not be a "([^"]*)"$`, iRunItWithANumberWhichShouldNotBeA)
}
```

**Reply by roskee** [reply]
That is right. but I think what @mfelipe wanted is a way to ignore this kind of syntax recommendation from `godog` in the first place. In my opinion, `number1` should just be treated as a part of the whole sentence. It is better to only parse the integer when it is on its own -i.e `number 1`.

**Reply by mfelipe** [reply]
Yes, that's also an option, but I want to be able to control the snippet generation.
